File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
       HashMap<Integer,Integer> map = new HashMap<>();

       for(int i=0;i<nums.length;i++){
        int sum = target - nums[i];
        if(map.containsKey(sum)){
            return new int[]{map.get(sum),i};
        }
        map.put(nums[i],i);
       }

       return new int[]{-1,-1};
    }
}

// sum = 9 - 2 = 7
// map.add(2,1);
// sum = 9 - 7 = 2
// map.contains(2) --> 0,1

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}     

        for i,num in enumerate(nums):

            sum = target - num 

            if sum in map:      
                return [map[sum],i]

            map[num] = i   

        return []    